### PR TITLE
Remove deprecated restricted lab option

### PR DESCRIPTION
**Pull request by BillCarsonFr**
Removes the deprecated use experimental restricted join rule lab option

### CODE REVIEW

**Review by bmarty — CHANGES_REQUESTED** [review]
Please do not touch the translated strings, I do not want to have conflict with Weblate.
See https://github.com/vector-im/element-android/blob/main/CONTRIBUTING.md#removing-existing-strings for more details

**Comment by BillCarsonFr** [comment]
> Please do not touch the translated strings, I do not want to have conflict with Weblate. See https://github.com/vector-im/element-android/blob/main/CONTRIBUTING.md#removing-existing-strings for more details

Updated

**Review by bmarty — CHANGES_REQUESTED** [review]
See my remark also the CI is not happy:
`w: /workdir/vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt: (58, 55): Parameter 'vectorPreferences' is never used`

File: vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt [hunk 1]
```diff
@@ -72,7 +72,7 @@ class CreateRoomViewModel @AssistedInject constructor(@Assisted private val init
         val restrictedSupport = session.getHomeServerCapabilities().isFeatureSupported(HomeServerCapabilities.ROOM_CAP_RESTRICTED)
         val createRestricted = when (restrictedSupport) {
             HomeServerCapabilities.RoomCapabilitySupport.SUPPORTED          -> true
-            HomeServerCapabilities.RoomCapabilitySupport.SUPPORTED_UNSTABLE -> vectorPreferences.labsUseExperimentalRestricted()
+//            HomeServerCapabilities.RoomCapabilitySupport.SUPPORTED_UNSTABLE -> vectorPreferences.labsUseExperimentalRestricted()
```
[the remark this answers is not in the document]
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomViewModel.kt]: yeah, I hesitated a bit but will remove now (done)

File: vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt [hunk 2]
```diff
@@ -75,7 +75,7 @@ class RoomSettingsViewModel @AssistedInject constructor(@Assisted initialState:
         val restrictedSupport = homeServerCapabilities.isFeatureSupported(HomeServerCapabilities.ROOM_CAP_RESTRICTED)
         val couldUpgradeToRestricted = when (restrictedSupport) {
             HomeServerCapabilities.RoomCapabilitySupport.SUPPORTED          -> true
-            HomeServerCapabilities.RoomCapabilitySupport.SUPPORTED_UNSTABLE -> vectorPreferences.labsUseExperimentalRestricted()
+//            HomeServerCapabilities.RoomCapabilitySupport.SUPPORTED_UNSTABLE -> vectorPreferences.labsUseExperimentalRestricted()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt:78]: Still a change to do here

Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/app/features/roomprofile/settings/RoomSettingsViewModel.kt]: thx, updated (+ rebase)

**Review by bmarty — APPROVED** [review]
Thanks for the update. I will squash and merge.

File: changelog.d/4889.removal [hunk 3]
```diff
@@ -0,0 +1 @@
+Remove deprecated experimental restricted space lab option
```
Comment (bmarty) [inline changelog.d/4889.removal:1]: FYI: using `removal` extension is reserved for changes in the SDK. For this change `misc` would have been better. (I am seeing that when doing the release, so I will update the changelog manaully, no worries)

**Pull request merged**